src/psyclone/psyir/tools/dependency_tools.py

```diff
@@ -273,102 +224,443 @@ def _is_loop_suitable_for_parallel(self, loop, only_nested_loops=True):
+
+        :param comp_ind1: component_indices of the first array access.
+        :type comp_ind1:  \
+            :py:class:`psyclone.core.component_indices.ComponentIndices`
+        :param comp_ind2: component_indices of the first array access.
+        :type comp_ind2:  \
+            :py:class:`psyclone.core.component_indices.ComponentIndices`
+
+        :return: partition information.
+        :rtype: list of 2-tuple of set-of-str and list of integer
+
+        '''
+        # Get the (string) name of all variables used in each subscript
+        # of the two accesses. E.g. `a(i,j+k)` --> [ {"i"}, {"j","k"}]
+        set_of_loop_vars = set(loop_vars)
+        indices_1 = DependencyTools.get_flat_indices(comp_ind1,
+                                                     set_of_loop_vars)
+        indices_2 = DependencyTools.get_flat_indices(comp_ind2,
+                                                     set_of_loop_vars)
+        # This list stores the partition information, which
+        # is a pair consisting of:
+        # - a set of all loop variables used in the subscript of
+        #   both accesses
+        # - list of all subscripts. Initially these lists contain
+        #   only one subscript, but they will be modified later
+        # Example: `a(i,j)` and `a(i,k)` -->
+        #          [ ({"i"}, [(0,0)]), ({"j","k"}, [(0,1)])]
+        partition_infos = []
+        for i, indx in enumerate(comp_ind1.iterate()):
+            partition_infos.append((indices_1[i].union(indices_2[i]), [indx]))
+
+        # Check each loop variable to find subscripts in which they are used:
+        for loop_var in loop_vars:
+            first_use = None
+            # The partition_infos list will get modified inside the loop.
+            # So we use a while loop to accommodate this.
+            k = 0
+            while k < len(partition_infos):
+                part_info = partition_infos[k]
+                if loop_var not in part_info[0]:
+                    k += 1
+                    continue
+                # Current loop variable is used in subscript k.
+                # If this is the first usage of this loop variable,
+                # just remember this index and continue with next partition.
+                if first_use is None:
+                    first_use = k
+                    k += 1
+                    continue
+                # Partition `k` and `first_use` share a variable.
+                # Add partition info from `k` to `first_use`, and
+                # then discard partition `k`.
+                partition_infos[first_use] = \
+                    (partition_infos[first_use][0].union(part_info[0]),
+                     partition_infos[first_use][1] + part_info[1])
+                del partition_infos[k]
+
+        return partition_infos
+
+    # -------------------------------------------------------------------------
+    @staticmethod
+    def independent_0_var(index_exp1, index_exp2):
```

Comment:
I am not sure what "0_var" means in the name. Could this be named "are_indenpendent"?

But also, this method that is only used once and it is equivalent to:
 `SymbolicMaths.get().never_equal(index_exp1, index_exp2)`
maybe we don't need this whole method if there is the appropriate comment?